site/content/languages/zig.md

```diff
@@ -0,0 +1,213 @@
++++
+title = "Zig"
++++
+
+## Introduction
+
+Beginning with 0.4.0 [Zig][1] can generate `%.wasm` files instead of
+architecture-specific binaries through three targets:
+
+* `wasm32-emscripten`: mostly for browser (JavaScript) use.
+* `wasm32-freestanding`: for standalone use in or outside the browser.
+* `wasm32-wasi`: for use outside the browser.
+
+This document is maintained by wazero, which is a WebAssembly runtime that
+embeds in Go applications. Hence, our notes focus on targets used outside the
+browser, tested by wazero: `wasm32-freestanding` and `wasm32-wasi`.
+
+## Overview
+
+When Zig compiles a `%.zig` file with a `wasm32-*` target, the output `%.wasm`
+depends on a subset of features in the [WebAssembly 2.0 
+Core specification]({{< ref "/specs#core" >}}) and [WASI]({{< ref "/specs#wasi" >}}) host
+functions.
+
+Unlike some compilers, Zig also supports importing custom host functions and
+exporting functions back to the host.
+
+Here's a basic example of source in Zig:
+
+```zig
+export fn add(a: i32, b: i32) i32 {
+    return a + b;
+}
+```
+
+The following is the minimal command to build a Wasm file.
+
+```bash
+zig build-lib main.zig -target wasm32-freestanding -dynamic
```

Comment:
```suggestion
zig build-lib -dynamic main.zig -target wasm32-freestanding
```

let's group "build-lib -dynamic" together so that this lines with the below.

Reply:
I'll group all the switches together:

```
zig build-lib -dynamic -target wasm32-freestanding main.zig
```